### ISSUE TITLE
bug(mat-option): aria-selected is always false when used inside multiselect mat-autocomplete

### ISSUE THREAD

**Issue by tHe-AK** [angular/components#31431]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

I tried creating mat-autocomplete with multiselect options along with a checkbox following a StackOverflow answer: 
https://stackoverflow.com/a/50569422/8043493
The functionality works fine, but the `aria-selected` attribute for the selected option is `false`. I tried setting it explicitly using `attr.aria-selected`, but it is not working.

### Reproduction

![Image](https://github.com/user-attachments/assets/e9d45eb5-557d-4401-99fa-1a564f133a77)

StackBlitz link: https://stackblitz.com/edit/w8j5bi2o
Steps to reproduce:
1. Select an option from the autocomplete.
2. Navigate to the same option with a screen reader opened, OR check the aria-selected attribute value for the selected mat-option.

### Expected Behavior

The selected `mat-option` should have `aria-selected` as `true`.

### Actual Behavior

The selected `mat-option` has `aria-selected` as `false`.

### Environment

- Angular: 15.0.5
- CDK/Material: 15.2.9
- Browser(s): Edge
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
Closing since this isn't a configuration we officially support.
